### ISSUE TITLE
SQS ChangeMessageVisibility from within Lambda function

### ISSUE THREAD

**Issue by cjroebuck** [localstack/localstack#2039]
sqs.changeMessageVisibility according to [this](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ChangeMessageVisibility.html#API_ChangeMessageVisibility_Examples) makes a request directly to the Queue's URL rather than via the SQS Endpoint. (I'm already setting SQS endpoint to LOCALSTACK_HOSTNAME - makes no difference for this call)..

In localstack the queue URL host returned is by default set to the value of HOSTNAME_EXTERNAL, which I'm setting to localstack, but calling this method within a lambda, does not work and I get the error:

```
UnknownEndpoint: Inaccessible host: `localstack'. This service may not be available in the `us-east-1' region. 
```

So, how can I make the queue URL accessible from within a lambda using localstack? Effectively I need to make **http://localstack** accessible from within lambda.

I'm using the following cmd to start localstack:

```
LAMBDA_EXECUTOR=docker-reuse DEBUG=1 LAMBDA_REMOTE_DOCKER=0 DATA_DIR=/tmp/localstack/data HOSTNAME_EXTERNAL=localstack localstack start
```

**Comment by cjroebuck** [comment]
I managed to workaround this by setting HOSTNAME_EXTERNAL="192.168.65.2" but the next issue I have is that when calling the changeMessageVisibility function of the message from within the lambda function (using javascript sdk), it requires the receiptHandle of the message. It seems that the receiptHandle of the sqs messages are all hardcoded to 'MessageReceiptHandle' [here](https://github.com/localstack/localstack/blob/81a409be4e8602c47923711cf752c13cfcf49507/localstack/services/awslambda/lambda_api.py#L283) causing the following error from sqs:

```
InvalidParameterValue: InvalidParameterValue; see the SQS docs.
```
I'm guessing this means this kind of functionality isnt supported by localstack currently?

For completeness this is what I was trying to get working: 

SNS topic -> SQS queue -> triggers lambda -> if lambda fails, call changeMessageVisibility of the SQS message to implement an exponential backoff retry mechanism..

**Comment by whummer** [comment]
Thanks for reporting @cjroebuck . This should be fixed in #2145 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
